tcms/testcases/views.py

```diff
@@ -457,14 +461,15 @@ def get_selected_testcases(request):
     Arguments:
     - request: REQUEST object.
     """
-    REQ = request.POST or request.GET
-    if REQ.get('selectAll', None):
+    method = request.POST or request.GET
+    if method.get('selectAll', None):
         plan = plan_from_request_or_none(request)
         cases, _search_form = query_testcases_from_request(request, plan)
         return cases
-    else:
-        pks = [int(pk) for pk in REQ.getlist('case')]
-        return TestCase.objects.filter(pk__in=pks)
+    keys = []
+    for key in method.getlist('case'):
+        keys.append(int(key))
+    return TestCase.objects.filter(pk__in=keys)
```

Comment:
If `method.getlist('case')` returns a list of strings that represent primary keys then `.filter(pk__in=...)` should work just fine with it. Slit this change into a separete pull request, create a test for it and try to remove the intermediate for loop.